src/core/main.c

```diff
@@ -1364,7 +1363,6 @@ int main(int argc, char *argv[]) {
 
                 /* Running outside of a container as PID 1 */
                 arg_system = true;
-                make_null_stdio();
```

Comment:
Agreed. I don't know if by "host case" you included systemd user instance, but that would make sense to connect stdios to the null device for all cases (system, user, containers), no ?